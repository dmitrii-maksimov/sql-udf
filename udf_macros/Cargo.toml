[package]
name = "udf-macros"
version = "0.1.2"
edition = "2021"
description = "UDF procedural macros implementation"
repository = "https://github.com/pluots/udf/tree/main/udf_macros"
license = "Apache-2.0"
# autotests = false

[lib]
proc-macro = true

[dependencies]
heck = "0.4.0"
proc-macro2 = "1.0.42"
quote = "1.0.21"
syn = { version = "1.0.99", features = ["full", "extra-traits"] }

[dev-dependencies]
trybuild = { version = "1.0.65", features = ["diff"] }

[package.metadata.release]
shared-version = true

# [[test]]
# name = "tests"
# path = "tests/harness.rs"

# [dev-dependencies]
# syn = { version = "1.0.99", features = ["full", "extra-traits"] }
