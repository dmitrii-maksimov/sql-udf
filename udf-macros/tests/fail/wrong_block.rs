//! Registration should fail no anything that is not an impl

use udf_macros::register;

// Registration is not allowed on non-impls
#[register]
struct X {}

fn main() {}
