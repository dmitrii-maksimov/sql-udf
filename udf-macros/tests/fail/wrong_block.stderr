error: expected `impl`
 --> tests/fail/wrong_block.rs:7:1
  |
7 | struct X {}
  | ^^^^^^
